tee-worker/litentry/core/assertion-build/src/dynamic/contracts/DynamicAssertion.sol

```diff
@@ -113,40 +83,72 @@ abstract contract DynamicAssertion {
         return (value);
     }
 
-    function concatenateStrings(string memory a, string memory b)
-    internal
-    pure
-    returns (string memory)
-    {
-        bytes memory concatenatedBytes = abi.encodePacked(a, b);
-        return string(concatenatedBytes);
+    function GetString(
+        string memory url,
+        string memory jsonPointer,
+        HttpHeader[] memory headers
+    ) internal returns (string memory returnVal) {
+        bytes memory encoded = abi.encode(url, jsonPointer, headers);
+        uint256 encoded_len = encoded.length;
+
+        assembly {
+            if iszero(call(not(0), 0x03EA, 0, add(encoded, 0x20), encoded_len, returnVal, 0x82)) {
```

Comment:
`0x32` means that only 32 bytes of returned string will be read from result of this call, remaining will be omitted